src/cogs/dynamic_load.py

```diff
@@ -57,23 +62,45 @@ def _fmt_cog_list(self, input_list: list) -> str:
         ret = "\n".join(f"- {i}" for i in input_list)
         return f"```\n{ret}\n```"
 
-    @commands.command(name="dloader")
+    @commands.hybrid_command(name="dloader")
+    @app_commands.autocomplete(cog_name=_entry_autocomplete)
     async def entry(self, context, cog_name: str):
         self.logging.info(f"entry called with {cog_name}")
 
         if cog_name == "all":
             reloaded = self._fmt_cog_list(await self._reload_all_cogs())
+            self.logging.info(f"Reloaded\n{reloaded}.")
             await context.send(f"Reloaded\n{reloaded}")
 
         elif cog_name == "list":
             resp = self._fmt_cog_list(self.bot.extensions.keys())
             await context.send(f"Cogs currently loaded:\n{resp}")
 
+        elif cog_name == "tree":
+            if (str(context.message.author.id) == self.bot.admin_user):
+                self.bot.tree.copy_global_to(guild = context.guild)
+                await self.bot.tree.sync(guild = context.guild)
```

Comment:
I don't understand what these things are doing. I get that `gtree` is likely `guild tree` but an inline comment explaining this would be useful.

Reply:
Part of the issue is that I also don't fully understand what they are doing 😅

Reply:
@fjebaker 
For more clarity,
I had a hard time getting the slash commands to work, even when following online advice and tutorials. In the end I used both what `tree` and `gtree` do, not sure in what order, so I don't understand fully.
However, as of 23c4697f57fbfd6d90e0d33023a5c187de1149ea I've renamed them to `globaltree` and `guildtree`.
My (probably wrong) understanding of it is:
- `globaltree` updates the global configuration for the command tree for the bot, including what slash commands can be called, what arguments those slash commands can have, any input validation / autosuggest for those arguments, etc. But, it can take a while for the global configuration update to be reflected on all servers
- `guildtree` copies the global configuration to the configuration of a single guild (i.e the one that the command is called on), and then synchronises for that guild, which is a much faster configuration change and so the update will be on that server faster
There's also apparently heavy rate limiting, so these should only be called if there is definitely a change in the command tree structures that needs to be updated, hence why these are privileged commands.